Tools/scripts/build_binaries_history.py

```diff
@@ -87,3 +87,66 @@ def record_run(self, hash, tag, start_time, duration):
                   "values (?, ?, ?, ?)",
                   (hash, tag, start_time, duration))
         c.commit()
+
+    def generate_firmware_size_json(self, json_filepath, githash):
+        c = self.conn()
+        sql_query = "SELECT * FROM build WHERE hash = ?"
+        sql_result = c.execute(sql_query, (githash,))
+        records = sql_result.fetchall()
+        # print("Total rows are:  ", len(records))
+        c.close()
+        sql_dict = {
+            "hash": 0,
+            "tag": 1,
+            "vehicle": 2,
+            "board": 3,
+            "frame": 4,
+            "text": 5,
+            "data": 6,
+            "bss": 7,
+            "start_time": 8,
+            "duration": 9,
+        }
+        for row in records:
+            vehicle = row[sql_dict["vehicle"]]
+            board = row[sql_dict["board"]]
+            frame = row[sql_dict["frame"]]
+            text = row[sql_dict["text"]]
+            data = row[sql_dict["data"]]
+            bss = row[sql_dict["bss"]]
+
+            if vehicle == "Rover":
+                vehicle = "ardurover"  # match binary name on build/board/bin
+            if frame is not None:
+                vehicle = vehicle + "-" + frame
+            build_json = [{
+                board.lower(): {vehicle.lower(): {"text": text, "data": data, "bss": bss}},
+            }]
+            if json_filepath is not None:
```

Comment:
I think you wanted to check if this file exists.

Instead of doing all this, create an empty file outside the loop.  Perhaps check `len(records)` or similar first.

Reply:
no, we need to check that the file exists and isn't empty for the json lib to work